src/transformers/models/t5/convert_t5x_checkpoint_to_pytorch.py

```diff
@@ -0,0 +1,221 @@
+        k, o, q, v = t5x_attention_lookup(old, i, "decoder", "encoder_decoder_attention")
+        new[f"decoder.block.{i}.layer.1.layer_norm.weight"] = layer_norm
+        new[f"decoder.block.{i}.layer.1.EncDecAttention.k.weight"] = k.T
+        new[f"decoder.block.{i}.layer.1.EncDecAttention.o.weight"] = o.T
+        new[f"decoder.block.{i}.layer.1.EncDecAttention.q.weight"] = q.T
+        new[f"decoder.block.{i}.layer.1.EncDecAttention.v.weight"] = v.T
+
+        # Block i, layer 2 (MLP).
+        layer_norm = t5x_layer_norm_lookup(old, i, "decoder", "pre_mlp_layer_norm")
+        wi, wo = t5x_mlp_lookup(old, i, "decoder", split_mlp_wi)
+        new[f"decoder.block.{i}.layer.2.layer_norm.weight"] = layer_norm
+        if split_mlp_wi:
+            new[f"decoder.block.{i}.layer.2.DenseReluDense.wi_0.weight"] = wi[0].T
+            new[f"decoder.block.{i}.layer.2.DenseReluDense.wi_1.weight"] = wi[1].T
+        else:
+            new[f"encoder.block.{i}.layer.2.DenseReluDense.wi.weight"] = wi.T
+        new[f"decoder.block.{i}.layer.2.DenseReluDense.wo.weight"] = wo.T
+
+    new["decoder.final_layer_norm.weight"] = old["decoder/decoder_norm/scale"]
+    new["decoder.block.0.layer.0.SelfAttention.relative_attention_bias.weight"] = old[
+        "decoder/relpos_bias/rel_embedding"
+    ].T
+
+    # LM Head (only in v1.1 checkpoints, in v1.0 embeddings are used instead)
+    if "decoder/logits_dense/kernel" in old:
+        new["lm_head.weight"] = old["decoder/logits_dense/kernel"].T
+
+    return new
+
+
+def make_state_dict(converted_params):
+    """Prepares a state dict for the PyTorch model."""
+    # Make a state dict with torch tensors.
+    state_dict = collections.OrderedDict([(k, torch.from_numpy(v.copy())) for (k, v) in converted_params.items()])
+
+    # Add what is missing.
+    if "encoder.embed_tokens.weight" not in state_dict:
+        state_dict["encoder.embed_tokens.weight"] = state_dict["shared.weight"]
+
+    if "decoder.embed_tokens.weight" not in state_dict:
+        state_dict["decoder.embed_tokens.weight"] = state_dict["shared.weight"]
+
+    if "lm_head.weight" not in state_dict:  # For old 1.0 models.
+        print("Using shared word embeddings as lm_head.")
+        state_dict["lm_head.weight"] = state_dict["shared.weight"]
+
+    return state_dict
+
+
+def load_t5x_weights_in_t5(model, config, t5x_checkpoint_path):
+    """Replaces the params in model witht the T5X converted params."""
+    num_weights = sum([1 for _ in model.named_parameters()])
+    num_layers = config.num_layers
+
+    variables = checkpoints.load_t5x_checkpoint(t5x_checkpoint_path)
+    converted = convert_t5x_to_pytorch(variables, num_layers=num_layers)
+    if len(converted) != num_weights:
+        raise RuntimeError(
+            f"Number of converted parameters {len(converted)} does not match "
+            f"number of original parameters {num_weights}"
+        )
```

Comment:
Small nit : you probably don't need this as `model.load_state_dict(state_dict, strict=True)` will give you more details on which keys were not properly converted and which keys are missing 😉

Reply:
Fair, I'll remove this. It was more of a sanity check while coding up the script :)